Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -796,11 +813,107 @@ namespace xr
             }
         }
 
+        void UpdatePlanes(std::map<NativePlaneIdentifier, Plane*>& existingPlanes, std::vector<Plane>& newPlanes, std::vector<Plane*>& deletedPlanes)
+        {
+            if (!IsTracking())
+            {
+                return;
+            }
+
+            // First check if any existing planes have been subsumed by another plane, if so add them to the list of deleted planes
+            CheckForSubsumedPlanes(existingPlanes, deletedPlanes);
+
+            // Next check for updated planes, and update their pose and polygon or create a new plane if it does not yet exist.
+            ArFrame_getUpdatedTrackables(session, frame, AR_TRACKABLE_PLANE, trackableList);            
+            int32_t size{};
+            ArTrackableList_getSize(session, trackableList, &size);
+            for (int i = 0; i < size; i++)
+            {
+                // Get the plane.
+                ArTrackable* trackable;
+                ArTrackableList_acquireItem(session, trackableList, i, &trackable);
+                auto planeTrackable = reinterpret_cast<ArPlane*>(trackable);
+
+                // Check if this plane has been subsumed. If so skip it as we are about to delete this plane.
+                ArPlane* subsumingPlane = nullptr;
+                ArPlane_acquireSubsumedBy(session, planeTrackable, &subsumingPlane);
+                if (subsumingPlane != nullptr)
+                {
+                    ArTrackable_release(trackable);
+                    ArTrackable_release(reinterpret_cast<ArTrackable*>(subsumingPlane));
+                    continue;
+                }
+
+                // Get the center pose.
+                float rawPose[7]{};
+                ArPlane_getCenterPose(session, planeTrackable, tempPose);
+                ArPose_getPoseRaw(session, tempPose, rawPose);
+
+                // Dynamically allocate the polygon array, and fill it in.
+                int32_t polygonSize;
+                ArPlane_getPolygonSize(session, planeTrackable, &polygonSize);
+                float* polygon = reinterpret_cast<float*>(malloc(sizeof(float) * polygonSize));
+
+                // Get the polygon.
+                ArPlane_getPolygon(session, planeTrackable, polygon);
+
+                // Update the existing plane if it exists, otherwise create a new plane, and add it to our list of planes.
+                auto planeIterator = existingPlanes.find(reinterpret_cast<NativePlaneIdentifier>(trackable));
```

Comment:
It would be a bit easier to follow if this used `planeTrackable` instead of `trackable`.
```suggestion
                auto planeIterator = existingPlanes.find(reinterpret_cast<NativePlaneIdentifier>(planeTrackable));
```